core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7716BuildDeadlock.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.maven.it;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+
+import org.apache.maven.shared.verifier.Verifier;
+import org.apache.maven.shared.verifier.util.ResourceExtractor;
+import org.junit.jupiter.api.Test;
+
+/**
+ * This is a test set for
+ * <a href="https://issues.apache.org/jira/browse/MNG-7716">MNG-7716</a>.
+ * Executing the project should not deadlock
+ *
+ */
+class MavenITmng7716BuildDeadlock
+    extends AbstractMavenIntegrationTestCase
+{
+
+    public MavenITmng7716BuildDeadlock()
+    {
+        super( "[3.8.0,)"  );
+    }
+
+    /**
+     * Verify that maven invocation works (no NPE/error happens).
+     *
+     * @throws Exception in case of failure
+     */
+    @Test
+    void testSingleMojoNoPom()
+        throws Exception
+    {
+        File testDir = ResourceExtractor.simpleExtractResources( getClass(), "/mng-7716" );
+
+        Verifier verifier = newVerifier( testDir.getAbsolutePath() );
+        verifier.addCliArgument( "install" );
+        verifier.addCliArgument( "-f" );
+        verifier.addCliArgument( "settings" );
+        verifier.execute();
+        verifier.verifyErrorFreeLog();
+        Verifier verifier2 = newVerifier( testDir.getAbsolutePath() );
+        verifier2.addCliArgument( "-T1C" );
+        verifier2.addCliArgument( "versions:set" );
```

Comment:
Versions Maven Plugin must be declared and int boostrap to be availabe offline

Reply:
How do I do that?

Reply:
Fixed